Add new football.json dataset (#146)
vega/vega-datasets: sources.md

Transformed using /scripts/flights.js

## `football.json`

> Football match outcomes across multiple divisions. This dataset is a subset of a larger dataset found at https://github.com/openfootball/football.json

Reviewer: ```suggestion
Football match outcomes across multiple divisions. This dataset is a subset of a larger dataset from https://github.com/openfootball/football.json.
```
